ee/query-service/app/api/api.go

```diff
@@ -0,0 +1,124 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"go.signoz.io/signoz/ee/query-service/dao"
+	"go.signoz.io/signoz/ee/query-service/interfaces"
+	"go.signoz.io/signoz/ee/query-service/license"
+	baseapp "go.signoz.io/signoz/pkg/query-service/app"
+	baseint "go.signoz.io/signoz/pkg/query-service/interfaces"
+	rules "go.signoz.io/signoz/pkg/query-service/rules"
+	"go.signoz.io/signoz/pkg/query-service/version"
+)
+
+type APIHandlerOptions struct {
+	DataConnector  interfaces.DataConnector
+	AppDao         dao.ModelDao
+	RulesManager   *rules.Manager
+	FeatureFlags   baseint.FeatureLookup
+	LicenseManager *license.Manager
+}
+
+type APIHandler struct {
+	opts APIHandlerOptions
+	baseapp.APIHandler
+}
+
+// NewAPIHandler returns an APIHandler
+func NewAPIHandler(opts APIHandlerOptions) (*APIHandler, error) {
+
+	baseHandler, err := baseapp.NewAPIHandler(baseapp.APIHandlerOpts{
+		Reader:       opts.DataConnector,
+		AppDao:       opts.AppDao,
+		RuleManager:  opts.RulesManager,
+		FeatureFlags: opts.FeatureFlags})
+
+	if err != nil {
+		return nil, err
+	}
+
+	ah := &APIHandler{
+		opts:       opts,
+		APIHandler: *baseHandler,
+	}
+	return ah, nil
+}
+
+func (ah *APIHandler) FF() baseint.FeatureLookup {
+	return ah.opts.FeatureFlags
```

Comment:
Opts is added to reduce parameters for NewAPIHandler() call. 

FF() is utility function that adds concise-ness and a central place for accessing feature flags. It's better to write 

`ah.FF().CheckFeature(model.SSO) ` than 

`ah.Opts.FeatureFlags.CheckFeature(model.SSO)`

This also helps keeping control of how feature flags are fetched at one place.

I realise the base API Handler still has vars featureFlags instead of opts.FeatureFlags. That could probably change to reduce duplication. 

The opts  is anyway going to carry the reference, so no point in keeping the same reference in base.ApiHandler